src/app/(main)/dashboard/components/voyage-dashboard/getDashboardData.ts

```diff
@@ -80,55 +132,112 @@ export const getDashboardData = async (
+    fetchFeatures,
+    user,
+    error,
+    teamId,
+    { teamId },
+  );
+  const projectIdeasResult = await fetchData<
+    IdeationData[],
+    { teamId: number }
+  >(fetchProjectIdeas, user, error, teamId, { teamId });
+  const techStackResult = await fetchData<TechStackData[], { teamId: number }>(
+    fetchTechStack,
+    user,
+    error,
+    teamId,
+    { teamId },
+  );
+  const resourcesResult = await fetchData<ResourceData[], { teamId: number }>(
+    fetchResources,
+    user,
+    error,
+    teamId,
+    { teamId },
+  );
+
+  if (typeof sprintsResult === "string")
+    return {
+      currentSprintNumber: null,
+      sprintsData: [],
+      user: null,
+      meetingsData: [],
+      voyageNumber: null,
+      voyageData: {} as Voyage,
+      features: [],
+      projectIdeas: [],
+      techStackData: [],
+      projectResources: [],
+      errorMessage: sprintsResult,
+      errorType: ErrorType.FETCH_SPRINT,
+    };
+
   let currentSprintNumber = null;
-  if (sprintsData.length > 0) {
-    const { number } = getCurrentSprint(sprintsData) as Sprint;
-    currentSprintNumber = number;
+  if (sprintsResult.sprintsData.length > 0) {
+    currentSprintNumber =
+      getCurrentSprint(sprintsResult.sprintsData)?.number ?? null;
   }
 
-  const meetingsData: {
-    title: string;
-    date: string;
-    link: string;
-    sprint: number;
-  }[] = [];
-
-  const fetchMeetingsPromises = sprintsData
-    .filter((sprint) => sprint.teamMeetings.length)
-    .map((sprint) =>
-      fetchMeeting({
-        sprintNumber: sprint.number,
-        meetingId: sprint.teamMeetings[0]?.id,
-      }),
-    );
+  const fetchMeetingsPromises = sprintsResult.sprintsData.map((sprint) =>
+    fetchMeeting({
+      sprintNumber: sprint.number,
+      meetingId: sprint.teamMeetings[0]?.id,
+    }),
+  );
 
   const fetchMeetingsResults = await Promise.all(fetchMeetingsPromises);
 
-  fetchMeetingsResults.forEach(([res, error]) => {
-    if (res) {
-      const { title, dateTime, meetingLink, sprint } = res;
-      const parsedDate = convertStringToDate(dateTime, user?.timezone ?? "");
-      const formattedDate = format(parsedDate, "yyyy-MM-dd h:mm a");
-      meetingsData.push({
-        title,
-        date: formattedDate,
-        link: meetingLink,
-        sprint: sprint.number,
-      });
-    } else if (error) {
-      errorMessage = error.message;
-      errorType = ErrorType.FETCH_MEETING;
-    }
-  });
+  const meetingsData = fetchMeetingsResults
+    .map(([res]) => {
+      if (res) {
+        const { title, dateTime, meetingLink, sprint } = res;
+        const parsedDate = convertStringToDate(dateTime, user?.timezone ?? "");
+        const formattedDate = format(parsedDate, "yyyy-MM-dd h:mm a");
+        return {
+          title,
+          date: formattedDate,
+          link: meetingLink,
+          sprint: sprint.number,
+        };
+      } else if (error) {
```

Comment:
@timothyrusso I think you're handling an error which comes through props from `getUser` request. In order to handle an error from`fetchMeeting` you need to use [res, error] instead of [res]. But then you'll get an error straight away because here:
```
const fetchMeetingsPromises = sprintsResult.sprintsData.map((sprint) =>
    fetchMeeting({
      sprintNumber: sprint.number,
      meetingId: sprint.teamMeetings[0]?.id,
    }),
  );
```
meetingId can be undefined (because teamMeetings can be an empty array) but you still make a request to a server and then get 404 error (just use `console.log(fetchMeetingsResults` and you'll see what I'm talking about in vs code terminal)), so in order to fix this behavior, you can just filter sprints  => `.filter((sprint) => sprint.teamMeetings.length)` first.

Reply:
Thanks @JaneMoroz , I've made an update let me know if it is correct

Reply:
Looks good!